sentry_sdk/integrations/django/caching.py

```diff
@@ -1,80 +1,138 @@
-    description = "{} ".format(method_name)
-
+def _get_key(args, kwargs):
+    # type: (Any, Any) -> str
     if args is not None and len(args) >= 1:
-        description += str(args[0])
+        return str(args[0])
     elif kwargs is not None and "key" in kwargs:
-        description += str(kwargs["key"])
+        return str(kwargs["key"])
+
+    return ""
 
+
+def _get_span_description(method_name, args, kwargs):
+    # type: (str, Any, Any) -> str
+    description = "{} {}".format(
+        method_name,
+        _get_key(args, kwargs),
+    )
     return description
 
 
-def _patch_cache_method(cache, method_name):
-    # type: (CacheHandler, str) -> None
+def _patch_cache_method(cache, method_name, address, port):
+    # type: (CacheHandler, str, Optional[str], Optional[str]) -> None
     from sentry_sdk.integrations.django import DjangoIntegration
 
     original_method = getattr(cache, method_name)
 
     @ensure_integration_enabled(DjangoIntegration, original_method)
-    def _instrument_call(cache, method_name, original_method, args, kwargs):
-        # type: (CacheHandler, str, Callable[..., Any], Any, Any) -> Any
+    def _instrument_call(
+        cache, method_name, original_method, args, kwargs, address, port
+    ):
+        # type: (CacheHandler, str, Callable[..., Any], Any, Any, Optional[str], Optional[str]) -> Any
+        integration = sentry_sdk.get_client().get_integration(DjangoIntegration)
+        is_set_operation = method_name.startswith("set")
+        is_get_operation = not is_set_operation
+
+        op = OP.CACHE_SET_ITEM if is_set_operation else OP.CACHE_GET_ITEM
         description = _get_span_description(method_name, args, kwargs)
 
-        with sentry_sdk.start_span(
-            op=OP.CACHE_GET_ITEM, description=description
-        ) as span:
+        with sentry_sdk.start_span(op=op, description=description) as span:
             value = original_method(*args, **kwargs)
 
-            if value:
-                span.set_data(SPANDATA.CACHE_HIT, True)
-
-                size = len(str(value))
-                span.set_data(SPANDATA.CACHE_ITEM_SIZE, size)
-
-            else:
-                span.set_data(SPANDATA.CACHE_HIT, False)
+            with capture_internal_exceptions():
+                if address is not None:
+                    # Strip eventually contained username and password in url style location
+                    if "://" in address:
+                        parsed_url = urlparse(address)
+                        address = "{}://{}{}".format(
+                            parsed_url.scheme or "",
+                            parsed_url.netloc or "",
+                            parsed_url.path or "",
+                        )
+
+                    span.set_data(SPANDATA.NETWORK_PEER_ADDRESS, address)
+
+                if port is not None:
+                    span.set_data(SPANDATA.NETWORK_PEER_PORT, int(port))
+
+                key = _get_key(args, kwargs)
```

Comment:
Can `key` ever be `None` in the args or kwargs? Asking because `str(None) == "None"`, so just wondering if this is a case we need to handle

Reply:
I do not think so. The only way it would be `"None"` would be if kwarg["key"] would return None (in _get_key()). And this code  is there since a very long time and no user has ever reported that they see `"None"` as the key.

Reply:
ah okay, did not realize this code was here before